Heat Equation 2D Bug Fixes

ikbuibui opened the pull request:
Fix race condition in image write
Improve stability condition
Dont validate for guard cells
Improve contrast in image write
Make image changes more obvious

psychocoderHPC commented:
@ikbuibui  I stopped the CI because jobs already failed

```
        for(uint32_t i = 1; i < extents[1] - 1; ++i)
                                ^~~~~~~
                                extent
/__w/alpaka/alpaka/example/heatEquation2D/src/analyticalSolution.hpp:33:21: note: 'extent' declared here
    T_Extent const& extent,
```

psychocoderHPC commented on example/heatEquation2D/src/analyticalSolution.hpp:
```diff
@@ -39,7 +39,7 @@ auto validateSolution(
     for(uint32_t j = 1; j < extent[0] - 1; ++j)
     {
-        for(uint32_t i = 0; i < extent[1]; ++i)
+        for(uint32_t i = 1; i < extents[1] - 1; ++i)
```
must be `extent`

ikbuibui replied:
fixed

ikbuibui committed:
Fix race condition in image write
Improve stability condition
Dont validate for guard cells
Improve contrast in image write
Make changes more obvious
